lib/request.js

```diff
@@ -1,18 +1,28 @@
 const http = require('http');
 const https = require('https');
+const zlib = require('zlib');
 
 module.exports = (options) => {
     options = Object.assign({
         protocol: 'https:'
     }, options);
     const httpModule = options.protocol === 'https:' ? https : http;
     return new Promise((resolve, reject) => {
-        const chunks = [];
         const request = httpModule.request(options, (response) => {
-            response.on('data', (chunk) => {
+            let bodyStream;
+            const chunks = [];
+            const encoding = response.headers && response.headers['content-encoding'];
+            if (encoding === 'gzip' || encoding === 'deflate') {
+                response.on('error', reject);
+                bodyStream = response.pipe(zlib.createUnzip());
+            } else {
+                bodyStream = response;
+            }
+            bodyStream.on('error', reject);
```

Comment:
However, `pipe` of Node.js document says:

>One important caveat is that if the Readable stream emits an error during processing, the Writable destination is not closed automatically. If an error occurs, it will be necessary to manually close each stream in order to prevent memory leaks.

I couldn't find a way to emit error on `response` except:

```js
response.emit('error', new Error('response error'));
response.pause();
```

But I also couldn't find memory leak with this case.